galaxy_api/api/ui/viewsets/namespace.py

```diff
@@ -75,6 +75,32 @@ def create(self, request, *args, **kwargs):
             return Response(serializer.data, status=status.HTTP_201_CREATED)
         return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
 
+    def update(self, request, *args, **kwargs):
+        groups = []
+        for account in request.data['groups']:
+            if account == RH_PE_ACCOUNT_SCOPE:
+                groups.append(account)
+            else:
+                group, _ = auth_models.Group.objects.get_or_create_identity(
+                        auth.RH_ACCOUNT_SCOPE, account.split(':')[1])
+                groups.append(group.name)
+        request.data['groups'] = groups
+
+        instance = self.get_object()
+        serializer = serializers.NamespaceUpdateSerializer(instance, data=request.data)
+        serializer.is_valid(raise_exception=True)
+        self.perform_update(serializer)
+
+        if getattr(instance, '_prefetched_objects_cache', None):
+            # If 'prefetch_related' has been applied to a queryset, we need to
+            # forcibly invalidate the prefetch cache on the instance.
+            instance._prefetched_objects_cache = {}
+
+        return Response(serializer.data)
+
+    def perform_update(self, serializer):
+        serializer.save()
+
```

Comment:
Just curious... when can a namespace be updated with a group, when that group has not yet been added to the database?

Reply:
@awcrosby this is the same workflow as if the user is authenticated for the first time and we still don't have the group stored in the database ([here](https://github.com/ansible/galaxy-api/blob/master/galaxy_api/auth/auth.py#L51)) — a new group record is created and assigned to the modified or created namespace object.

By not providing a way to assign group that's not tracked in our database, we're taking away partners ability to create namespaces on their own (missing the point of https://github.com/ansible/galaxy-dev/issues/176).

Reply:
Would be nice to have at least a way to verify, if the Red Hat Account with such an ID actually exists.

With this approach, we're just blindly relying on the partner to know his ID and that he provides it correctly.

Reply:
I think that if the group doesn't exist, the endpoint show throw an error saying it couldn't find the given group and return it to the user. That way if the user makes a typo when putting in the account ID, we aren't accidentally creating garbage accounts